botocore/credentials.py

```diff
@@ -275,7 +277,7 @@ class EnvProvider(CredentialProvider):
     # AWS_SESSION_TOKEN is what other AWS SDKs have standardized on.
     TOKENS = ['AWS_SECURITY_TOKEN', 'AWS_SESSION_TOKEN']
 
-    def __init__(self, environ=None, mapping=None):
+    def __init__(self, environ=None, mapping=None, profile_name=None):
```

Comment:
I think this is the wrong level of abstraction and couples profiles to retrieving environment variables when they don't need to be.  An env provider should just be a simple object that pulls credentials from environment variables and that's it.  If something up the call stack wants to either re-arrange cred providers or ignore what this class returns, that's another thing, but this class should just stay minimal.

It also gives this class weird semantics.  If you pass a profile_name to this class, then it won't do anything.